### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by Anny85-code**
### In this milestone we;
- [x] Refactored the main.rb file according to SOLID principle.
- [x] Refactored the app.rb file according to SOLID principle.

- [x] Refactored codes to adhere to YAGNI, KISS, and DRY rules and Ruby style guide .
  Thank you for reviewing! 👍

### CODE REVIEW

File: Classes/app.rb [hunk 1]
```diff
@@ -29,43 +47,33 @@ def create_person
...
-      puts 'Student doesnt have parent permission, cant rent books'
-    when 'y'
-      student = Student.new(age, name, parent_permission: false)
-      @people << student
-      puts 'Student created successfully'
-    end
+    permission = parent_permission == 'y'
+    student = Student.new(age, name, parent_permission: permission)
+    @people << student
+    puts 'Student created successfully'
```
Comment (mikethreels) [inline Classes/app.rb]: - [ ] in this method you are printing to the console, getting user input and saving the data, this is not in line with the single responsibility rule please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

File: Classes/app.rb [hunk 2]
```diff
@@ -29,43 +47,33 @@ def create_person
...
-    print 'Enter teacher age: '
-    age = gets.chomp.to_i
+  def create_teacher(age, name)
     print 'Enter teacher specialization: '
     specialization = gets.chomp
-    print 'Enter teacher name: '
-    name = gets.chomp
     teacher = Teacher.new(age, name, specialization)
     @people << teacher
     puts 'Teacher created successfully'
```
Comment (mikethreels) [inline Classes/app.rb]: - [ ] in this method you are printing to the console, getting user input and saving the data, this is not in line with the single responsibility rule please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

File: Classes/app.rb [hunk 3]
```diff
@@ -79,7 +87,7 @@ def list_all_people
     end
   end
 
-  def create_book()
+  def create_book
```
Comment (mikethreels) [inline Classes/app.rb:80]: - [ ] in this method you are printing to the console, getting user input and saving the data, this is not in line with the single responsibility rule please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi <STUDENT_MENTION>, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] the link to the license is not working as expected, please have another look at the [Microverse template](https://github.com/microverseinc/readme-template) for an example of how it is supposed to work.
- [ ] in the `create_rental` method you are printing to the console, getting user input and saving the data, this is not in line with the single responsibility rule please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [ ] [OPTIONAL] It might be a good idea to include all the collaborators in the readme file.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Anny85-code, @orozCoding AND @gtekle,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**